### PR TITLE
Fixed #1244, Spec 0.8.4 networking update

### LINKED ISSUE #1244: Spec v0.8.4 update - Networking
### What is wrong?

Mainly https://github.com/ethereum/eth2.0-specs/pull/1404

### PR DESCRIPTION
**Pull request by ChihChengLiang**
### What was wrong?

- Should merge after #1252

### How was it fixed?

- Address https://github.com/ethereum/eth2.0-specs/pull/1404
- [x] Rename Hello to Status
- [x] Respond the SSZ list request in chunks
- [x] Rename BeaconBlocks to BeaconBlocksByRange
- [x] Rename RecentBeaconBlocks to BeaconBlocksByRoot
- [ ] ~~Send goodbye as string not SSZ container.~~ Turns out the encoding is the same with or without the container

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: trinity/protocol/bcc_libp2p/utils.py [hunk 1]
```diff
@@ -345,55 +361,62 @@ def __init__(self, stream: INetStream):
...
-            self.peer_id,
-            to_formatted_dict(message),
-        )
+        self.debug(f"Respond {type(message).__name__}  {to_formatted_dict(message)}")
         await write_resp(self.stream, message, ResponseCode.SUCCESS)
 
+    async def write_chunk_response(self, messages: Sequence[MsgType]) -> None:
+        self.debug(f"Respond {len(messages)} chunks")
+        for message in messages:
+            await write_resp(self.stream, message, ResponseCode.SUCCESS)
```
Comment (NIC619) [inline trinity/protocol/bcc_libp2p/utils.py]: Should we handle `WriteMessageFailure` and break the loop?

**Comment by ChihChengLiang** [comment]
Thank you for the review @NIC619!

We now have 4 kinds of interactions:
1. `Interaction(stream)`: _handle_goodbye and say_goodbye
2. `handshake_interaction`: request_status and _handle_status
3. `post_handshake_handler_interaction`: _handle_beacon_blocks_by_range and _handle_beacon_blocks_by_root
4. `my_request_interaction`: request_beacon_blocks_by_range and request_beacon_blocks_by_root

We do `interaction.write_response` in 2 and 3,  and do `interaction.write_request` in 1, 2, and 4. 
- In 1 we handle MessageIOFailure (ReadMessageFailure and WriteMessageFailure) in the body of handle/say goodbye.
- In 2 we handle MessageIOFailure in the body of `new_handshake_interaction`
- In 3 we handle ReadMessageFailure and WriteMessageFailure in the body of `post_handshake_handler_interaction`
- In 4 we handle MessageIOFailure in `my_request_interaction` and reraise RequestFailure

File: trinity/protocol/bcc_libp2p/utils.py (hunk 1 above)
Comment (ChihChengLiang) [reply trinity/protocol/bcc_libp2p/utils.py]: WriteMessageFailure will be raised by write_resp, will break the loop, and will be handled by the caller of write_chunk_response.

**Comment by ChihChengLiang** [comment]
And we handle `PeerRespondedAnError` in `handshake_interaction ` and `my_request_interaction`. So that we can customize the handling of PeerRespondedAnError for different cases

**Comment by NIC619** [comment]
Yeah you are right. I completely forgot the different interactions used.🤦‍♂️

**Pull request merged**